app/models/concerns/foreman_puppet/extensions/host.rb

```diff
@@ -55,6 +55,15 @@ def search_by_config_group(_key, operator, value)
       end
 
       module PrependedMethods
+        def environment_id=(id)
+          self.environment = ForemanPuppet::Environment.find(id)
+          save
+        end
+
+        def inherited_attributes
```

Comment:
This method is missing for sure so that attributes from hostgroup are used. Same as in katello, openscap, salt...

Reply:
Do you know @ekohl (or know somehow you maybe can answer this question), why this method did not exist in puppet but is used for Katello, OpenSCAP, Salt. 
How does this work, if you create a host which is using a hostgroup, that this host is using the "templated values" from hostgroup without this code?

Reply:
As this is related to facets (environment_id is provided by the puppet facet), maybe we can also ping @ShimShtein